common/optimiser/Optimiser.java

```diff
@@ -83,67 +110,76 @@ public boolean isError() {
         return status == Status.ERROR;
     }
 
-    public synchronized void restartFromSolution() {
-        if (hasSolver) {
-            solver.reset();
-            setSolverModel();
-        } else initialiseSolver();
+    public void setConstraintsChanged() {
+        isConstraintsUpToDate = true;
```

Comment:
we allow `Constraint` to call back and notify the `Optimiser` that owns it that the constraint has changed

Reply:
Did you mean to set it to `false` here?

Reply:
oh shoot ahaha wow